### PR TITLE
virtualenvwrapper 4.8.4 (new formula)

### PR DESCRIPTION
**Pull request by georgexsh**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
new formula for python virtualenvwrapper.

### CODE REVIEW

**Comment by georgexsh** [comment]
```
==> brew audit virtualenvwrapper --online --new-formula
==> FAILED
virtualenvwrapper:
  * Formulae should not require patches to build. Patches should be submitted and accepted upstream first.
```

no idea how to deal with this issue, the patch is targeted to homebrew only, upstream will not accept such patch.

**Comment by SMillerDev** [comment]
They already have a way to set the python: `$VIRTUALENVWRAPPER_PYTHON`. The formula should install a wrapper script that sets this.

File: Formula/virtualenvwrapper.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    sha256 "665e48dd54c84b98b71a657acb49104c54e7652bce9c1c4f6c6976ed4c827a29"
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  def caveats
+    <<~EOS
+      To activate virtualenvwrapper, add the following at the end of your #{shell_profile}:
```
Comment (SMillerDev) [inline Formula/virtualenvwrapper.rb:59]: That's not a solution. Brew should be setting this environment with something like `write_env_script`

Comment (georgexsh) [reply Formula/virtualenvwrapper.rb:59]: cool, will rewrite with`write_env_script`

Comment (georgexsh) [reply Formula/virtualenvwrapper.rb:68]: `write_env_script` use `exec`, but `virtualenvwrapper.sh` is a bash script, loads by the user with `source`, so I use `write` directly.

File: Formula/virtualenvwrapper.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      #!/bin/sh
+      export VIRTUALENVWRAPPER_PYTHON="#{libexec}/bin/python"
+      source "#{libexec}/bin/virtualenvwrapper.sh"
+    SH
+  end
+
+  def caveats
+    <<~EOS
+      To activate virtualenvwrapper, add the following at the end of your #{shell_profile}:
+        source virtualenvwrapper.sh
```
Comment (SMillerDev) [inline Formula/virtualenvwrapper.rb:69]: Will it look for this in the path?

Comment (georgexsh) [reply Formula/virtualenvwrapper.rb:69]: I guess "it" refers to `bash`, then answer is yes:

bash:
```
source: source filename [arguments]
    Execute commands from a file in the current shell.

    Read and execute commands from FILENAME in the current shell.  The
    entries in $PATH are used to find the directory containing FILENAME.
    If any ARGUMENTS are supplied, they become the positional parameters
    when FILENAME is executed.
```

zsh:
```
 source file [ arg ... ]
        Same as `.', except that the current directory is always searched and  is  always  searched  first,
        before directories in $path.
```

Comment (SMillerDev) [reply Formula/virtualenvwrapper.rb:69]: Yeah, I was refering to the `source` command. But I guess that's built-in for all shells. Either way, no objection from me then.

**Review by SMillerDev — APPROVED** [review]

**Comment by georgexsh** [comment]
@SMillerDev may I know how to proceed this PR?

**Review by nandahkrishna — APPROVED** [review]
Thanks, @georgexsh.

**Pull request closed without merging**